abi/src/main/java/org/web3j/abi/TypeDecoder.java

```diff
@@ -354,7 +605,10 @@ static BigInteger asBigInteger(Object arg) {
             return Numeric.toBigInt((String) arg);
         } else if (arg instanceof byte[]) {
             return Numeric.toBigInt((byte[]) arg);
-        } else if (arg instanceof Double || arg instanceof Float) {
+        } else if (arg instanceof Double
+                || arg instanceof Float
+                || arg instanceof java.lang.Double
+                || arg instanceof java.lang.Float) {
```

Comment:
Is this meant to be checking both primitive and complex types? If so should the first two not be lower case?